### PR TITLE
balena-action-hup-utils: bump to 4.0.0 to allow dev upgrades

### PR DESCRIPTION
**Pull request by xginn8**
HQ: https://github.com/balena-io/balena/issues/1815
HQ: https://github.com/balena-io/balena/issues/1322

Change-type: minor
Signed-off-by: Matthew McGinn <matthew@balena.io>

### CODE REVIEW

File: tests/integration/models/device.spec.coffee [hunk 1]
```diff
@@ -2087,7 +2087,43 @@ describe 'Device Model', ->
...
+					['Resin OS 2.0.0+rev3 (prod)', 'prod']
+					['Resin OS 2.0.0+rev3 (prod)', '']
+					['Resin OS 2.0.0+rev5 (prod)', '']
+					['Resin OS 2.0.2+rev2', 'prod']
+					['Resin OS 2.0.3+rev1', 'prod']
+					['Resin OS 2.0.4+rev1', 'prod']
+					['Resin OS 2.0.5', 'prod']
+					['Resin OS 2.0.5+rev1', 'prod']
+					['Resin OS 2.0.6+rev1', 'prod']
+					['Resin OS 2.0.6+rev2', 'prod']
```
Comment (sradevski) [inline tests/integration/models/device.spec.coffee]: We can drop some of these as there is a lot of redundancy.

Comment (xginn8) [reply tests/integration/models/device.spec.coffee]: Removed some similar cases.

**Comment by xginn8** [comment]
@thgreasi looks like we're hitting the same error as in https://github.com/balena-io-modules/balena-hup-action-utils/pull/19:

```
Error: could not get uid/gid

[ 'nobody', 0 ]
```

**Comment by thgreasi** [comment]
@xginn8 we don't run the sdk tests on balenaCI yet :(
So you should only need appveyor to pass, which it seems that had errors while running the node tests but worked for the browser ones. So, I just restarted them since it seems like an API hickup.

**Pull request merged**